fix: show more help text and version info in det-deploy [DET-4373]

mackrorysd opened the pull request:
## Description

Things like `det-deploy gcp` or `det-deploy aws` just print errors because the code assumes you have either an environment AND command or neither. Now requiring the command before going into the environment code. This prompts the user to try --help, and I've added help text for the local_state_path flags on GCP. I also wanted a --version flag while I was debugging this to see if I was actually on a release version, or a development version, etc...

## Test Plan

I ran all combinations mentioned above.

davetroiano reviewed (commented):
@mackrorysd could you add `--local-state-path` to `docs/how-to/installation/gcp.txt` and `docs/how-to/installation/aws.txt`?

davetroiano reviewed (approved):
tested this out and lgtm -- approving but requesting the docs update

davetroiano commented:
one other thing I noticed from CI is that this needs some formatting fix to pass linting (`make check-deploy`)

mackrorysd commented:
> @mackrorysd could you add `--local-state-path` to `docs/how-to/installation/gcp.txt` and `docs/how-to/installation/aws.txt`?

The only change to aws.txt is that we'll show help text if you specify the environment but no command (instead of a runtime error for accessing another arg that hasn't had default values applied yet). I've updated gcp.txt with the --local-state-path flag though.

> one other thing I noticed from CI is that this needs some formatting fix to pass linting (make check-deploy)

I fixed the formatting w/ black, however locally mypy still complains about `required` being an unexpected keyword arg to add_subparsers. However it is documented and changes the functionality for me as expect, so I don't know what it's on about: https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers

mackrorysd commented:
Ah - looks like required=True was added in Python 3.7. mypy . --python-version 3.6 fails for me like it does here. --python-version 3.7 succeeds. We already require Python 3.7, right? If so, I'll just add the Python version to the Makefile.

mackrorysd commented:
Correction to previous comments: we require Python 3.5 (for the CLI) or Python 3.6 (for contributing). Looks like someone in local already figured this out: you can just add required=True to the result of the add_subparsers call. Works on Python <3.7, still eliminates the bug where you end up entering code before the parser has fully run.
